aiohttp/client_ws.py

```diff
@@ -351,14 +361,21 @@ async def receive(self, timeout: Optional[float] = None) -> WSMessage:
                 return WS_CLOSED_MESSAGE
             except WebSocketError as exc:
                 self._close_code = exc.code
-                await self.close(code=exc.code)
+                await self._close_from_receive(exc.code)
                 return WSMessageError(data=exc)
             except Exception as exc:
                 self._exception = exc
                 self._set_closing()
                 self._close_code = WSCloseCode.ABNORMAL_CLOSURE
-                await self.close()
+                await self._close_from_receive(WSCloseCode.OK)
```

Comment:
This is a bit unexpected, but this is how is currently works. Maybe something to reconsider later.